pandas/core/construction.py

```diff
@@ -412,6 +412,9 @@ def sanitize_array(
 
     # GH#846
     if isinstance(data, np.ndarray):
+        if copy is None:
+            # copy by default for DataFrame({"A": ndarray})
+            copy = True
```

Comment:
if you make this False i would be ok with this; True is a breaking change

Reply:
On master, we copy for dict of ndarrays

```python
In [4]: import numpy as np

In [5]: import pandas as pd
a
In [6]: a = np.array([1, 2])

In [7]: df = pd.DataFrame({"A": a})

In [8]: df.iloc[0, 0] = 10

In [9]: a
Out[9]: array([1, 2])
```

Perhaps it's not clear from the diff, but this section is hit only for dict inputs. We continue to not copy by default for ndarray `data` (on this branch):

```python
In [4]: a = np.array([1, 2])

In [5]: df = pd.DataFrame({"A": a})

In [6]: df.iloc[0, 0] = 10

In [7]: a
Out[7]: array([1, 2])

In [8]: b = np.ones((4, 4))

In [9]: df = pd.DataFrame(b)

In [10]: df.iloc[0, 0] = 10

In [11]: b
Out[11]:
array([[10.,  1.,  1.,  1.],
       [ 1.,  1.,  1.,  1.],
       [ 1.,  1.,  1.,  1.],
       [ 1.,  1.,  1.,  1.]])
```

Reply:
Although, now that I see your objection, other uses of `sanitize_array` will need to be checked. I've only updated `DataFrame.__init__`, but I'll need to ensure other places use copy=False if they were relying on the default.